pkg/apis/networking/types.go

```diff
@@ -48,31 +48,35 @@ const (
 
 // NetworkPolicySpec provides the specification of a NetworkPolicy
 type NetworkPolicySpec struct {
-	// Selects the pods to which this NetworkPolicy object applies. The array of
-	// ingress rules is applied to any pods selected by this field. Multiple network
-	// policies can select the same set of pods. In this case, the ingress rules for
-	// each are combined additively. This field is NOT optional and follows standard
+	// Selects the pods to which this NetworkPolicy object applies for ingress and/or egress
+	// (as determined by `policyTypes`).  The relevant one or ones of `ingress` and `egress`
+	// list connections that this policy allows to be made to/from the selected pods.
+	// Multiple NetworkPolicy objects can select a given pod, and the connections they
+	// allow are combined additively.
+	// If any NetworkPolicy applies to a given pod for ingress then (a) the default is
+	// to allow ingress to the pod only from the pod's node and (b) the applicable policies
+	// add allowed ingress to the pod; otherwise all ingress to the pod is allowed.
+	// If any NetworkPolicy applies to a given pod for egress then (a) the default is
+	// to deny all egress from the pod and (b) the applicable policies add allowed egress from the pod;
+	// otherwise all egress from the pod is allowed.
+	// A connection from some source pod to some destination pod is allowed if egress from the source
```

Comment:
Maybe instead say “read the NetworkPolicy documentation to understand more about this API”?

Reply:
This PR _does_ put such a remark on the Spec field of NetworkPolicy.  But we also typically document the semantics in comments in the API file, right?  I think what's here is an accurate statement of the semantics of this API.  The website doc explains it more slowly and carefully and with examples.

Reply:
I'm neutral on this really. When it renders into https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/ it's going to look kind-of readable but also a bit like a wall of text.

(What I think I really want is Markdown doc comments in Golang, but that's a lot of abstraction away from the issue at hand).

So, no objection.

Reply:
Our API godocs should ALWAYS clearly define the semantics of the API.  "Read the docs" should be for examples, advanced use cases.  Put simply, if a user can't understand what the API does from this godoc, we either need to add more, or clarify more.  Any important implication should be captured in the API.

If that's not clear from API review guidelines, I'll take a todo to make it that way.

Reply:
Also, we do want to make sure that clean paragraphs can show up from godoc into api doc, if that's not happening that's a bug / area we can improve.